packages/builder/src/actions/projects.ts

```diff
@@ -439,52 +447,93 @@ export const fetchProjectApplications =
             }
           }
           `,
-            chain.id,
-            {
-              projectID: projectApplicationID,
-            },
-            reactEnv
-          );
-
-          if (response.errors) {
-            datadogRum.addError(response.error, { projectID });
-            console.error(response.error);
-            return [];
-          }
-
-          const applications = response.data.roundApplications.map(
-            (application: any) => ({
-              status: convertStatusToText(application.status),
-              roundID: application.round.id,
-              chainId: chain.id,
-              metaPtr: application.metaPtr,
-            })
-          );
-
-          if (applications.length === 0) {
-            return [];
-          }
+          chain.id,
+          {
+            projectID: projectApplicationID,
+            roundID,
+          },
+          reactEnv
+        );
 
-          dispatch({
-            type: PROJECT_APPLICATIONS_LOADED,
-            projectID,
-            applications,
-          });
+        if (response.errors) {
+          datadogRum.addError(response.error, { projectID });
+          console.error(response.error);
+          return [];
+        }
 
-          return applications;
-        } catch (error: any) {
-          datadogRum.addError(error, { projectID });
-          console.error(error);
+        const applications = await filterLegitApplications(
+          response.data.roundApplications,
+          chain.id
+        );
 
+        if (applications.length === 0) {
           return [];
         }
-      })
+
+        return applications;
+      } catch (error: any) {
+        datadogRum.addError(error, { projectID });
+        console.error(error);
+
+        return [];
+      }
+    })
+  );
+
+  return apps.flat();
+};
+
+export const fetchProjectApplicationInRound = async (
+  applicationId: string,
+  roundID: string,
+  reactEnv?: any
+): Promise<any> => {
+  const splitApplicationId = applicationId.split(":");
+  const projectChainId = Number(splitApplicationId[0]);
+  const projectRegistryAddress = splitApplicationId[1];
+  const projectNumber = splitApplicationId[2];
+
+  const projectApplicationID = generateUniqueRoundApplicationID(
```

Comment:
```suggestion
  const projectApplicationId = generateUniqueRoundApplicationId(
```

Reply:
The problem: we use sometimes Id and sometimes ID across the apps. Changing it at one place results in a change in other files. would keep it for now and maybe add it to our refactor list